python/mlc_chat/quantization/fp8_quantization.py

```diff
@@ -393,16 +425,17 @@ def forward(self, x: nn.Tensor, indptr: nn.Tensor) -> nn.Tensor:  # pylint: disa
         w_format = self.weight_dtype.split("_")[0]
 
         ### TODO: enable this for fp16 calibration
-        # if self.runtime == "max-calibration":
-        #     func = "cutlass.group_gemm_scale_fp16_sm90"
-        # else:
-        func = f"cutlass.group_gemm_{a_format}_{w_format}_fp16"
+        if self.runtime == "max-calibration":
+            func = "cutlass.group_gemm_scale_fp16_sm90"
+        else:
+            func = f"cutlass.group_gemm_{a_format}_{w_format}_fp16"
 
         if self.runtime == "cast":
-            func = func = "_host_scale"
+            func = func + "_host_scale"
             total_scale = 1.0
         else:
-            if self.weight_dtype == "e4m3_float8":
+            if self.runtime != "max-calibration" and self.weight_dtype == "e4m3_float8":
+                # for calibration, q_scale is already used to dequantize the weights
                 total_scale = local_scale * self.q_scale
```

Comment:
If we are using the existing GroupQuantize support for e4m3 weight quantization, `self.q_scale` will be a tensor of scales for each group. I think for this to work we would either need 
1. Support for whole tensor weight quantization so that self.q_scale is just a scalar
2. Support in the group gemm kernel for a vector of scales, with one group for all input features (per-channel)

I know we talked briefly about this yesterday but just to clarify are you planning to bring (1)? Or do you have a different thought?

Reply:
oh right, I missed this part, I'll change `GroupQuantize` to support (1)

Reply:
We decided this will be good in a follow up that @vinx13 already has in flight